.github/workflows/check-build-test.yml

```diff
@@ -71,71 +83,140 @@ jobs:
           name: ${{ github.job }}
           path: ${{ github.workspace }}/ccache-debug
 
-  sonar:
+
+  static-analysis:
     name: "Static Analysis"
-    if: |
-      github.event_name != 'pull_request' ||
-      github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name
     runs-on: ubuntu-latest
     container: docker.io/ogdf/clang:15
-    needs: [style, dirs, self-sufficiency, docs]
+    needs: [ style, dirs, self-sufficiency, docs ]
     env:
       CCACHE_COMPILERCHECK: "%compiler% -v"
     steps:
+      - name: Dump GitHub context
```

Comment:
I deduplicated the output so that its a little less overwhelming. As the info can still be helpful when debugging past failures (where you don't want to rerun the CI with some debugging options) and the output is not shown by default, I'd leave it in.